engine/consensus/matching/engine.go

```diff
@@ -158,6 +200,18 @@ func (e *Engine) loop() {
 	}
 }
 
+func (e *Engine) processFinalizationEvent() error {
+	finalized, err := e.state.Final().Head()
```

Comment:
This call here is really expensive, as it _must ready directly from the data base_. But we only use the ID for logging purpose. In `matching.Core`, we actually run a similar, equally expensive call: https://github.com/onflow/flow-go/blob/b7df2882af2a49525c8f3a901d1c02a6a474fcd1/engine/consensus/matching/core.go#L364-L365

A similar suggestion to [Leo's comment](https://github.com/onflow/flow-go/pull/887/files#r664968449):
* we could rename `matching.Core` method to `OnBlockFinalization()` _without_ any arguments.

Reply:
Good catch, implemented